src/helpers/turbo.ts

```diff
@@ -0,0 +1,87 @@
+import snapshot from '@snapshot-labs/snapshot.js';
+import fetch from 'node-fetch';
+import db from './mysql';
+
+const SCHNAPS_API_URL = process.env.SCHNAPS_API_URL;
+
+const RUN_INTERVAL = 10 * 1e3; // 10 seconds
+
+// Periodically sync the turbo status of spaces with the schnaps-api
+export async function trackTurboStatuses() {
+  if (SCHNAPS_API_URL) {
+    while (true) {
+      // Step 1: Query all the spaces from the schnaps-api
+      const spaces = await getSpacesExpirationDates();
+
+      // Step 2: Update the turbo status of the spaces in the database
+      updateTurboStatuses(spaces);
+
+      // Sleep for a while before running the loop again
+      await snapshot.utils.sleep(RUN_INTERVAL);
+    }
+  }
```

Comment:
```suggestion
  if (!SCHNAPS_API_URL) return;
  
  while (true) {
    // Step 1: Query all the spaces from the schnaps-api
    const spaces = await getSpacesExpirationDates();

    // Step 2: Update the turbo status of the spaces in the database
    updateTurboStatuses(spaces);

    // Sleep for a while before running the loop again
    await snapshot.utils.sleep(RUN_INTERVAL);
  }
```

Reply:
Reverted back to if in the condition. Not sure why (probably some weird JS things), but it wasn't working.
Yours: https://github.com/snapshot-labs/snapshot-sequencer/pull/523/commits/80636cfd5f45dd653da1c715d2d92db714b458d9

Revert: https://github.com/snapshot-labs/snapshot-sequencer/pull/523/commits/b169e07da67c6a30751c22041f93a62c9a9c5c9b

Reply:
Just tried, and it works. What issue are you seeing ?

a simple `console.log` inside the loop is printing

Reply:
I don't know, `yarn test` wasn't working (https://github.com/snapshot-labs/snapshot-sequencer/actions/runs/14125500679/job/39573608892) anymore so I guessed it had broken something. Since it's working in this version I didn't investigate further

Reply:
Fixed it https://github.com/snapshot-labs/snapshot-sequencer/actions/runs/14125500679/job/40103852183

Tests sometimes need to be retried, as some of them depends on external services, and can fail

Reply:
failing again : https://github.com/snapshot-labs/snapshot-sequencer/actions/runs/14310569196/job/40104292563